pkg/store/consul.go

```diff
@@ -91,11 +125,47 @@ func (s *Consul) Get(key string) (*KVPair, error) {
 }
 
 // Put a value at "key"
-func (s *Consul) Put(key string, value []byte) error {
-	p := &api.KVPair{Key: s.normalize(key), Value: value}
-	if s.client == nil {
-		log.Error("Error initializing client")
+func (s *Consul) Put(key string, value []byte, opts *WriteOptions) error {
+
+	key = s.normalize(key)
+
+	p := &api.KVPair{
+		Key:   key,
+		Value: value,
 	}
+
+	if opts != nil && opts.Ephemeral {
+		// Creates the global ephemeral session
+		// if it does not exist
+		if s.ephemeralSession == "" {
+			s.createEphemeralSession()
+		}
+
+		// Create lock option with the
+		// EphemeralSession
+		lockOpts := &api.LockOptions{
```

Comment:
Errr didn't answered the real question :D I think it's good here because we associate the key in `LockOptions` with the same global session. `LockOptions` should be different for each key though (to be associated with this global session).